File: docs-examples/example-python/src/main/python/micronaut/configuration/openapi/docs/security/OrderController.py
try:
    # tag::imports[]
    from micronaut.http.annotation import Controller, Get
    from micronaut.security.annotation import Secured
    from micronaut.security.rules import SecurityRule
    from io.swagger.v3.oas.annotations.security import *
    # end::imports[]
except ImportError:  # TODO(python): packages under `io.` other than `io.micronaut` cannot be imported at runtime
    from swagger.v3.oas.annotations.security import *


# tag::clazz[]
@Controller
@Secured(SecurityRule.IS_AUTHENTICATED)
class OrderController:

    @Get
    @SecurityRequirement(name="openid", scopes="openid")
    def index(self) -> str:
        return "Example Response"
# end::clazz[]

File: docs-examples/example-java/src/main/java/io/micronaut/configuration/openapi/docs/security/OrderController.java
package io.micronaut.configuration.openapi.docs.security;

// tag::imports[]
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.security.annotation.Secured;
import io.micronaut.security.rules.SecurityRule;
import io.swagger.v3.oas.annotations.security.SecurityRequirement;
// end::imports[]

// tag::clazz[]
@Controller
@Secured(SecurityRule.IS_AUTHENTICATED)
public class OrderController {

    @Get
    @SecurityRequirement(name = "openid", scopes = "openid")
    public String index() {
        return "Example Response";
    }
}
// end::clazz[]
